bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/LongPolling.java

```diff
@@ -166,8 +166,12 @@ public void onComplete(@Nullable Result result) {
                 // NOTE: This handler runs inside the HTTP thread, so we schedule the response
                 // handling in a new thread because the HTTP thread is terminated after the
                 // timeout expires.
-                scheduler.execute(() -> longPolling.onLongPollComplete(httpClient, subscriptionId, result,
-                        this.getContentAsString()));
+                scheduler.execute(() -> {
+                    String contentAsString = this.getContentAsString();
+                    if (contentAsString != null) {
```

Comment:
`contentAsString` will ultimately end up in `Gson::fromJSON` which should handle `null` gracefully (see https://github.com/google/gson/blob/99cc4cb11f73a6d672aa6381013d651b7921e00f/gson/src/main/java/com/google/gson/Gson.java#L1167).

I would therefore prefer to add `@Nullable` annotations to the `content` parameters of `onLongPollComplete()` and `handleLongPollResponse()` instead of changing the logic here.

Reply:
Agree, much better.